pkg/services/publicdashboards/api/api.go

```diff
@@ -126,7 +129,19 @@ func (api *Api) SavePublicDashboardConfig(c *models.ReqContext) response.Respons
 		PublicDashboard: pubdash,
 	}
 
-	pubdash, err := api.PublicDashboardService.SavePublicDashboardConfig(c.Req.Context(), &dto)
+	// Get the original dashboard
```

Comment:
We need to get the dashboard to validate whether it has template variables or not. What layer should be responsible for doing that?

I like the pattern of having the api layer be responsible for coordinating between different services in order to do the thing. But on the flipside, we don't want the api layer to become bloated with domain logic. Should the api layer do this coordination? Or should the publicdashboard service be responsible for calling other services that it needs?

Reply:
The public dashboard service has methods for retrieving a dashboard. I think we should do it there and then we can avoid using a dashboard query object and the dashboard service